### DISCUSSION TITLE
Existing Onboarding MR renovate/configure: Response code 409 (Conflict) Repository has unknown error: Another open merge request already exists for this source branch

### DISCUSSION THREAD

**Discussion by matzegebbe** [renovatebot/renovate#26262] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

RenovateBot/37.91.4

### If you're self-hosting Renovate, select which platform you are using.

gitlab.com

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

We had to update our access token a few weeks ago for Renovate used with Gitlab SaaS. Starting yesterday, with the latest version `RenovateBot/37.91.4`, we received the following error for all still open onboarding merge requests created by the fromer user (now `ghost user`)

```
HTTPError: Response code 409 (Conflict) Repository has unknown error: Another open merge request already exists for this source branch
```

**Deleting the Merge Request and branch `renovate/configure` solves the issue,** but it also appears that the new token/bot user is creating empty commits on this branch. 

We have many projects and some are still in the onboarding state. Is there a configuration that could ignore the existing branch or check beforehand if the branch is already there and, if yes, reuse it? I think a `RENOVATE_ONBOARDING_BRANCH` with a different name could also be a workaround. Has something changed in the GitLab API or Renovate here (as said, that error has been occurring for the past days)?

![image](https://github.com/renovatebot/renovate/assets/3482021/6f247c3b-58b6-4e59-b234-37cd12e3b7a1)
Compare with previous version is empty

minimal reproduce: To reproduce this on GitLab, let Renovate create an onboarding MR, then change the user/access token and rerun it.

All input is welcome, and thank you for your great product.

### Relevant debug logs

<details><summary>Logs</summary>

```
"message": "Response code 409 (Conflict)",
         "stack": "HTTPError: Response code 409 (Conflict)\n    at Request.<anonymous> (/opt/containerbase/tools/renovate/37.91.4/node_modules/got/dist/source/as-promise/index.js:118:42)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)",
         "options": {
           "headers": {
             "user-agent": "RenovateBot/37.91.4 (https://github.com/renovatebot/renovate)",
             "accept": "application/json",
             
             
 "response": {
           "statusCode": 409,
           "statusMessage": "Conflict",
           "body": {
             "message": [
               "Another open merge request already exists for this source branch: !1"
             ]
           },             

"msg": "Repository has unknown error"

```

</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by teefax** [comment]
We seem to have the same problem -- only difference is that we are using a self-hosted Gitlab instance.

**Comment by rarkins** [comment]
You mention that you needed to "update our access token" which on its own would imply same user, different token - but isn't it more correct to say that you needed to _change the user account which Renovate uses_?

If so then use https://docs.renovatebot.com/configuration-options/#ignoreprauthor

**Comment by matzegebbe** [comment]
Group access tokens in GitLab SaaS will result in what are called bot accounts - so yes, we have new user accounts. Great, we'll try this option - just a bit concerned about its impact on performance. https://docs.gitlab.com/ee/development/internal_users.html

**Reply by rarkins** [reply]
Changing Group access token (and therefore the "account") sounds like something which will happen quite regularly to many, so we should try to optimize for it. Unfortunately though it may not be possible to be efficient - we have no choice but to fetch all MRs and filter them inside Renovate.
